packages/SwingSet/src/kernel/deviceManager.js

```diff
@@ -55,33 +56,30 @@ export default function makeDeviceManager(
     deviceParameters,
   );
 
-  /**
-   * @typedef {['ok', CapData]} VatInvocationResults
-   */
-
-  /**
-   * @typedef {[string, string, CapData]} DeviceInvocation
-   * @property {string} 0 Kernel slot designating the device node that is the target of
-   * the invocation
-   * @property {string} 1 A string naming the method to be invoked
-   * @property {CapData} 2 A capdata object containing the arguments to the invocation
-   */
-
   /**
    * Invoke a method on a device node.
    *
-   * @param {DeviceInvocation} deviceInvocation
-   * @returns {VatInvocationResults} a VatInvocationResults object: ['ok', capdata]
-   * @throws {Error} an exeption if the invocation failed. This exception is fatal to
-   * the kernel.
+   * @param { DeviceInvocation } deviceInvocation
+   * @returns { DeviceInvocationResult }
    */
   function invoke(deviceInvocation) {
     const [target, method, args] = deviceInvocation;
-    const deviceResults = dispatch.invoke(target, method, args);
-    assert(deviceResults.length === 2);
-    assert(deviceResults[0] === 'ok');
-    insistCapData(deviceResults[1]);
-    return deviceResults;
+    try {
+      /** @type { DeviceInvocationResult } */
+      const deviceResults = dispatch.invoke(target, method, args);
+      // common error: raw devices returning capdata instead of ['ok', capdata]
+      assert.equal(deviceResults.length, 2, deviceResults);
```

Comment:
That's the message emitted if the assertion fails (or at least I think that's what it's for), so the developer at the console can figure out why it threw without going back and inserting a `console.log`. `optDetails` seems to be the official name. The default for `optDetails` reports the two values being compared, in this case "2" and whatever `.length` is. In the failure mode I ran into, where my raw device returned `{ body: .., slots: ..}` instead of `['ok', { body, slots }]`, the assertion printed `undefined !== 2`, which didn't help me very much.

Reply:
I also thought the 3rd arg was the message string, which is why I am puzzled because `deviceResults` is not a string.